js/src/components/google-combo-account-card/connected-google-combo-account-card.js

```diff
@@ -69,27 +81,87 @@ const ConnectedGoogleComboAccountCard = () => {
 		upsertAccount();
 	}, [ finalizeAdsAccountCreation, upsertAdsAccount, invalidateResolution ] );
 
-	if ( ! hasDetermined ) {
-		return <SpinnerCard />;
-	}
+	const handleCancelClick = () => {
+		setEditMode( false );
+	};
 
-	// @TODO: edit mode implementation in 2605
-	const editMode = false;
-	const shouldClaimGoogleAdsAccount = Boolean(
-		! loading && googleAdsAccount?.id && hasAccess === false
+	const handleEditClick = () => {
+		setEditMode( true );
+	};
+
+	// During MC account creation, we need to show the ConnectMC component
+	// when the account creation needs to show the Switch or Reclaim flow.
+	const googleMCHasError = [ 409, 403 ].includes(
+		resultCreateMCAccount.response?.status
 	);
 
-	const hasExistingGoogleMCAccounts = existingGoogleMCAccounts?.length > 0;
+	// After creating a new account, it may be connected but not ready
+	// (e.g., needing to reclaim the URL). In this case, we show the ConnectMC
+	// component, even if the existing accounts list has not yet updated.
 	const showConnectMC =
-		( editMode && hasExistingGoogleMCAccounts ) ||
-		( ! isGoogleMCReady && hasExistingGoogleMCAccounts );
-
-	const hasExistingGoogleAdsAccounts = existingGoogleAdsAccounts?.length > 0;
+		( googleMCHasError ||
+			hasGoogleMCConnection ||
+			hasExistingGoogleMCAccounts ) &&
+		( editMode || ! isGoogleMCReady );
+
+	// After creating a new account, it may not show up in the existing accounts list
+	// immediately. In this case, we show the ConnectAds component in edit mode unless
+	// we're showing the claim notice in the upper card.
 	const showConnectAds =
-		( ( editMode && hasExistingGoogleAdsAccounts ) ||
-			( ! isConnected && hasExistingGoogleAdsAccounts ) ) &&
+		( isConnected || hasExistingGoogleAdsAccounts ) &&
+		( editMode || ! isConnected ) &&
 		! shouldClaimGoogleAdsAccount;
 
+	// When Ads and MC are disconnected in edit mode, exit edit mode.
+	useEffect( () => {
+		if ( editMode && ! isGoogleMCReady && ! isConnected ) {
+			setEditMode( false );
+		}
+	}, [ editMode, isConnected, isGoogleMCReady ] );
```

Comment:
This doesn't work well when Google Ads account is waiting for claiming and Google Merchant Center account is not connected yet.

https://github.com/user-attachments/assets/80f49b3d-6195-4197-bfc4-5688e703f24a

Reply:
Good catch! [08f96a7](https://github.com/woocommerce/google-listings-and-ads/pull/2660/commits/08f96a74c81a35845c0a540e69f38939941ee639) should make this better.

Reply:
It's still not working well.

https://github.com/user-attachments/assets/fdde7fbd-9152-4151-a457-2db1da5e9e17